src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -203,5 +238,43 @@ private BoundNode CollectDynamicAnalysis(BoundStatement statement)
             
             return statementFactory.Block(ImmutableArray.Create(cellAssignment, statement));
         }
+
+        private static CSharpSyntaxNode SyntaxForSpan(BoundStatement statement)
```

Comment:
The actual spans have already deliberately diverged from sequence points, and will continue to diverge further as we add more dynamic analysis capabilities. However, integrating instrumentation and sequence point generation into a common architecture seems like a very good idea. There will probably be one or two more rounds of instrumentation fixes before we're in a position to pause and add some architecture for instrumentation during lowering.